adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -295,13 +296,30 @@ protected Router createRouter() {
         final Router router = Router.router(vertx);
         LOG.info("limiting size of inbound request body to {} bytes", getConfig().getMaxPayloadSize());
         router.route().handler(BodyHandler.create(DEFAULT_UPLOADS_DIRECTORY).setBodyLimit(getConfig().getMaxPayloadSize()));
+        Optional.ofNullable(getTenantTraceSamplingHandler())
+                .ifPresent(tenantTraceSamplingHandler -> router.route().handler(tenantTraceSamplingHandler)
```

Comment:
are you trying to insert this handler at a particular position? If so, IMHO you should make this explicit by setting the order ...

Reply:
This handler should be inserted directly after the BodyHandler one line above (otherwise there is an exception when processing requests). Since there is no order set for the BodyHandler, I haven't used one for the TenantTraceSamplingHandler either.